### PR TITLE
Clean-up Palette List code, part 2

### PR DESCRIPTION
**Pull request by felipemanga**
Further simplifies the palette list pop-up code, using smart pointers where possible.
Adds and makes use of the Thread Manager.

closes LibreSprite/LibreSprite-proposals#30

### CODE REVIEW

File: src/app/task_manager.h [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <unordered_set>
```
Comment (Zughy) [inline src/app/task_manager.h:17]: nitpicking: LibreSprite files have external libraries declared after the project headers

File: src/app/res/resources_loader.h [hunk 2]
```diff
@@ -1,41 +1,33 @@
...
-#include "base/concurrent_queue.h"
-#include "base/thread.h"
-#include "base/unique_ptr.h"
+#include <atomic>
+#include <memory>
 
-namespace app {
+#include "app/res/resource.h"
+#include "app/task_manager.h"
+#include "base/injection.h"
```
Comment (Zughy) [inline src/app/res/resources_loader.h:15]: nitpicking: includes order

File: src/app/ui/palette_listbox.cpp [hunk 3]
```diff
@@ -5,14 +5,12 @@
 // it under the terms of the GNU General Public License version 2 as
 // published by the Free Software Foundation.
 
+#include "doc/palette.h"
```
Comment (Zughy) [inline src/app/ui/palette_listbox.cpp:8]: does this have to be called before the config.h?

File: src/app/ui/palette_listbox.cpp [hunk 4]
```diff
@@ -30,17 +28,13 @@ using namespace skin;
 
 class PaletteListItem : public ListItem {
 public:
-  PaletteListItem(PaletteResource* palResource) :
-    ListItem(palResource->name()), m_palResource(palResource) {}
+  PaletteListItem(std::shared_ptr<doc::Palette> palette, const std::string& name) :
+    ListItem(name),
+    m_palette(palette) {}
```
Comment (Zughy) [inline src/app/ui/palette_listbox.cpp:33]: nitpicking: it bugs me a bit that the order is palette -> name, but then it becomes name -> palette

Comment (felipemanga) [reply src/app/ui/palette_listbox.cpp:33]: A base class has to be initialized before the members.
The order a constructor accepts arguments and the order it initializes things are not supposed to be related.

File: src/app/ui/palette_listbox.cpp (hunk 3 above)
Comment (felipemanga) [reply src/app/ui/palette_listbox.cpp]: nope

**Review by Zughy — APPROVED** [review]
Works

**Pull request merged**